src/Components/HomeArticles.jsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from "react";
+import { getAllArticles } from "./api";
+import { ArticleCard } from "./ArticleCard";
+import { Link } from "react-router-dom";
+
+export const HomeArticles = () => {
+  const [articles, setArticles] = useState([]);
+
+  const [articlesToShow, setArticlesToShow] = useState(4);
+
+  const updateArticlesToShow = () => {
+    const windowWidth = window.innerWidth;
+    if (windowWidth < 768) {
+      setArticlesToShow(2);
+    } else if (windowWidth < 1024) {
+      setArticlesToShow(3);
+    } else {
+      setArticlesToShow(4);
+    }
+  };
+
+  useEffect(() => {
+    getAllArticles(articlesToShow, 1)
+      .then((data) => {
+        setArticles(data);
+      })
+      .catch((error) => {
+        console.log("FetchError", error);
+      });
+
+    window.addEventListener("resize", updateArticlesToShow);
```

Comment:
I don't believe this is working as you'd hoped as your page is not responsive

Reply:
Actually When I reduce my with to 1024 at home page it displays 3 cards in box less then 768px it displays 2 cards.